In [1]:
import pandas as pd

### Step 1: Information about the NOC version 1.3 update Dec2019
in this step we will uste the information from the official website and the NOC we will use 2 files for this process. in this steps is mandatory to have internet connection in order to download the oficial documents

In [3]:
noc_structure = pd.read_csv("https://www.statcan.gc.ca/eng/statistical-programs/document/noc-cnp-2016-structure-v4-eng.csv")

In [14]:
noc_structure.shape

(690, 5)

In [15]:
noc_structure.sample(10)

Level Hierarchical structure  Code  \
185      3            Minor group   221   
183      4             Unit group  2175   
232      4             Unit group  3111   
366      4             Unit group  5244   
514      4             Unit group  7322   
178      3            Minor group   217   
581      3            Minor group   841   
314      4             Unit group  4217   
518      4             Unit group  7333   
476      4             Unit group  7245   

                                           Class title  \
185         Technical occupations in physical sciences   
183                       Web designers and developers   
232                              Specialist physicians   
366                         Artisans and craftspersons   
514                       Motor vehicle body repairers   
178     Computer and information systems professionals   
581  Mine service workers and operators in oil and ...   
314                        Other religious occupations   
518                               Electrical mechanics   
476          Telecommunications line and cable workers   

                                      Class definition  
185  This minor group includes chemical, geological...  
183  Web designers and developers research, design,...  
232  Specialists in clinical medicine diagnose and ...  
366  Artisans and craftspersons use manual and arti...  
514  Motor vehicle body repairers repair and restor...  
178  This minor group includes professionals workin...  
581  This minor group includes underground mine ser...  
314  Workers in other religious occupations include...  
518  Electrical mechanics maintain, test, rebuild a...  
476  Telecommunications line and cable workers inst...

In [4]:
noc_elements = pd.read_csv("https://www.statcan.gc.ca/eng/statistical-programs/document/noc-cnp-2016-element-v4-eng.csv")

In [13]:
noc_elements.shape

(43126, 9)

In [16]:
noc_elements.sample(19)

Level  Code                                        Class title  \
42216      4  9614       Labourers in wood, pulp and paper processing   
30916      4  8241                        Logging machinery operators   
20775      4  6316                         Other services supervisors   
1756       4   412  Government managers - economic analysis, polic...   
35841      4  9422              Plastics processing machine operators   
39898      4  9525  Assemblers, fabricators and inspectors, indust...   
13197      4  3215                    Medical radiation technologists   
21981      4  6523                  Airline ticket and service agents   
37729      4  9445                    Fabric, fur and leather cutters   
9129       4  2173                   Software engineers and designers   
15390      4  4154               Professional occupations in religion   
36483      4  9434            Other wood processing machine operators   
5970       4  1411                     General office support workers   
149        4    12           Senior government managers and officials   
16349      4  4169  Other professional occupations in social scien...   
31089      4  8252  Agricultural service contractors, farm supervi...   
31434      4  8412  Oil and gas well drilling and related workers ...   
21513      4  6411  Sales and account representatives - wholesale ...   
42919      4  9619  Other labourers in processing, manufacturing a...   

      Element Type Label English  \
42216               All examples   
30916    Employment requirements   
20775               Exclusion(s)   
1756                Exclusion(s)   
35841    Employment requirements   
39898    Illustrative example(s)   
13197               All examples   
21981               All examples   
37729    Employment requirements   
9129                All examples   
15390    Illustrative example(s)   
36483               All examples   
5970                 Main duties   
149                  Main duties   
16349               All examples   
31089                Main duties   
31434               All examples   
21513    Employment requirements   
42919               All examples   

                             Element Description English  Unnamed: 5  \
42216                                   log deck tender          NaN   
30916  Experience requirements vary depending on the ...         NaN   
20775  Managers in customer and personal services, n....         NaN   
1756   Labour policy analysts  (See 4164 Social polic...         NaN   
35841  Completion of a statistical process control (S...         NaN   
39898  switchgear and control panel assembler - indus...         NaN   
13197                       nuclear medicine technician          NaN   
21981           client service representative - airline          NaN   
37729  Some secondary school education is usually req...         NaN   
9129                           software design engineer          NaN   
15390                                              imam          NaN   
36483              drum barker tender - wood processing          NaN   
5970   Process incoming and outgoing mail, manually o...         NaN   
149    Establish objectives for the organization in a...         NaN   
16349                       archaeological field worker          NaN   
31089               May supervise general farm workers.          NaN   
31434                pumpman/woman - oil field services          NaN   
21513  Experience in sales or in an occupation relate...         NaN   
42919         general labourer - light-fixture assembly          NaN   

       Unnamed: 6  Unnamed: 7  Unnamed: 8  
42216         NaN         NaN         NaN  
30916         NaN         NaN         NaN  
20775         NaN         NaN         NaN  
1756          NaN         NaN         NaN  
35841         NaN         NaN         NaN  
39898         NaN         NaN         NaN  
13197         NaN         NaN         NaN  
21981         NaN         NaN   

the elements csv file has 9 columns and 43,126 rows and the Stucture file has 5 columns and 690 rows. from the sample exploration, the elements table contains null columns that will be remove before insertion to postgrest


In [18]:
noc_elements = noc_elements.dropna(1)

In [19]:
noc_elements.shape

(43126, 5)